### PR TITLE
Enable data loading

### PR DESCRIPTION
**Pull request by jnwltr**
- serialize game data to local storage every month of simulation
- store list of saved games to local storage
- save and load data for chart labels
- add dynamic game speed for testing

### CODE REVIEW

File: frontend/src/app/services/mitigations.service.ts [hunk 1]
```diff
@@ -188,8 +188,10 @@ export class MitigationsService {
       .subscribe(diff => this.set(diff.newValue));
 
     this.setValue$
-      .pipe(untilDestroyed(this))
-      .subscribe(preset => this.set(preset[1]));
+      .pipe(
+        startWith(['reset', gameService.game.mitigations] as PresetValue),
```
Comment (jnwltr) [inline frontend/src/app/services/mitigations.service.ts:192]: @JaroslavSoldat @Petr26 do you think this might cause any problem?

Comment (JaroslavSoldat) [reply frontend/src/app/services/mitigations.service.ts:192]: It is ok IMO.
Isn't it duplicated by gameService.reset$ below?
Never mind, rather twice than not set.

File: frontend/src/app/game/game.service.ts [hunk 2]
```diff
@@ -141,14 +146,86 @@ export class GameService {
...
+    return this.httpClient.post('/api/game-data', gameData).pipe(
+      tap((data: any) => this.saveGameId(data.id)),
+    );
+  }
+
+  private saveGameId(id: string) {
+    const storageValue = window.localStorage.getItem(LocalStorageKey.SAVED_GAME_IDS);
+    let ids = [];
+
+    if (storageValue) ids = JSON.parse(storageValue);
```
Comment (JaroslavSoldat) [inline frontend/src/app/game/game.service.ts:168]: Consider
`const ids = storageValue ? JSON.parse(storageValue) : [];`

File: frontend/src/app/game/game.service.ts [hunk 3]
```diff
@@ -141,14 +146,86 @@ export class GameService {
...
+
+    ids.push(id);
+    window.localStorage.setItem(LocalStorageKey.SAVED_GAME_IDS, JSON.stringify(ids));
+  }
+
+  private saveCheckpoint() {
+    window.localStorage.setItem(LocalStorageKey.LAST_GAME_DATA, JSON.stringify(this.getGameData()));
+  }
+
+  private loadGameFromJson() {
```
Comment (JaroslavSoldat) [inline frontend/src/app/game/game.service.ts:178]: Consider just `loadGame()` ... compare `saveGameId()`, `saveCheckpoint()`

**Review by JaroslavSoldat — APPROVED** [review]

**Pull request merged**